src/components/carousel/__tests__/index.spec.tsx

```diff
@@ -36,41 +32,43 @@ const Page = ({children, ...others}:{children: React.ReactNode, others?: any}) =
 };
 
 describe('Carousel render tests', () => {
-  afterEach(() => CarouselDriver.clear());
-
   describe('initialPage', () => {
     it('should be set to default initialPage', async () => {
-      const driver = new CarouselDriver({component: <TestCase/>, testID});
+      const renderTree = render(<TestCase/>);
+      const driver = CarouselDriver({renderTree, testID});
 
       expect((await driver.getContentOffset()).x).toBe(0);
     });
 
     it('should be set to initialPage = 2', async () => {
-      const driver = new CarouselDriver({component: <TestCase initialPage={2}/>, testID});
+      const renderTree = render(<TestCase initialPage={2}/>);
+      const driver = CarouselDriver({renderTree, testID});
 
       expect((await driver.getContentOffset()).x).toBe(Constants.screenWidth * 2);
     });
   });
 
   describe('onScroll', () => {
     it('should trigger onScroll from the second scroll', async () => {
-      const driver = new CarouselDriver({component: <TestCase/>, testID});
+      const renderTree = render(<TestCase/>);
+      const driver = CarouselDriver({renderTree, testID});
 
-      await driver.scroll(Constants.screenWidth); //NOTE: first scroll doesn't fire onScroll
+      await driver.scroll({contentOffset: {x: Constants.screenWidth}}); //NOTE: first scroll doesn't fire onScroll
```

Comment:
I just realized that you are receiving `props` in this method which the usage is more complex than it should be.
This will be much easier:
```
      await driver.scroll({x: Constants.screenWidth}, lessCommonOptionObject); // don't change the name for the second param :)
```